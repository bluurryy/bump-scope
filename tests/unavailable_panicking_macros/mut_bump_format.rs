use bump_scope::{Bump, MutBumpString, mut_bump_format};

pub fn greet<'a>(bump: &'a mut Bump, greeting: &str, name: &str) -> MutBumpString<&'a mut Bump> {
    mut_bump_format!(in bump, "{greeting}, {name}!")
}

fn main() {}
