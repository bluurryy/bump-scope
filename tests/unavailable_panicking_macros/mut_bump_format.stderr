error: the potentially panicking api of `mut_bump_format!` is not available
       help: enable `bump-scope`'s "panic-on-alloc" feature or use `try`:
       `mut_bump_format!(try in bump, "{greeting}, {name}!")`
 --> tests/unavailable_panicking_macros/mut_bump_format.rs:4:5
  |
4 |     mut_bump_format!(in bump, "{greeting}, {name}!")
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `$crate::__mut_bump_format_panic_on_alloc` which comes from the expansion of the macro `mut_bump_format` (in Nightly builds, run with -Z macro-backtrace for more info)
